### PR TITLE
Redis - Add support for the probabilistic data structures

### PR DESCRIPTION
**Pull request by cescoffier**
This PR adds the commands from the Redis-Bloom module (probabilistic data structures).
It provides support for:

- Bloom filters
- Cuckoo filters
- Count-min sketches
- Top-K sketches

_What about T-digest_? The `t-digest` group is not implemented as it does not seem to be provided by the Redis stack at the moment (while the commands are listed on the doc).

### CODE REVIEW

File: extensions/redis-client/runtime/src/main/java/io/quarkus/redis/runtime/datasource/ReactiveTransactionalCountMinCommandsImpl.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
+package io.quarkus.redis.runtime.datasource;
+
+import java.util.List;
+import java.util.Map;
+
+import io.quarkus.redis.datasource.countmin.*;
```
Comment (machi1990) [inline extensions/redis-client/runtime/src/main/java/io/quarkus/redis/runtime/datasource/ReactiveTransactionalCountMinCommandsImpl.java:6]: nit: remove usage of wildcard import?

**Review by machi1990 — APPROVED** [review]
lgtm

Thanks for adding tests along side this feature.

Left one nitpick.

File: docs/src/main/asciidoc/redis-reference.adoc [hunk 2]
```diff
@@ -278,6 +278,10 @@ As mentioned above, the API is divided into groups:
 - string - `.value(valueType)`
 - transactions - `withTransaction`
 - json - `.json()` (requires the https://redis.com/modules/redis-json/[redis-json] module on the server side)
+- bloom - `.bloom()` (requires the https://redis.com/modules/redis-bloom/[redis-bloom] module on the server side)
+- cuckoo - `.cuckoo()` (requires the https://redis.com/modules/redis-bloom/[redis-bloom] module on the server side, which also provides the cuckoo filter commands)
+- count-min - `.countmin()` (requires the https://redis.com/modules/redis-bloom/[redis-bloom] module on the server side, which also provides the count-min filter commands)
+- top-k - `.topk()` (requires the https://redis.com/modules/redis-bloom/[redis-bloom] module on the server side, which also provides the top-k filter commands)
```
Comment (machi1990) [inline docs/src/main/asciidoc/redis-reference.adoc:284]: It'll be very useful to show example usages of these but probably something for the followup PR

Comment (cescoffier) [reply docs/src/main/asciidoc/redis-reference.adoc:284]: I would rather redirect to the Redis documentation. 
Before using these commands (and associated structures), you must have some ideas and mathematical knowledge. Quarkus documentation should not become a math lesson. 

There are some basic usage examples in the test. I can copy snippets, but again, that won't be very useful if you don't understand the structure. (add items, execute some operations and receives opaque results)

File: extensions/redis-client/runtime/src/main/java/io/quarkus/redis/runtime/datasource/ReactiveTransactionalCountMinCommandsImpl.java (hunk 1 above)
Comment (cescoffier) [reply extensions/redis-client/runtime/src/main/java/io/quarkus/redis/runtime/datasource/ReactiveTransactionalCountMinCommandsImpl.java]: Fixed (and a few others). This is due to the generator I used.... Need to fix it :-D

File: docs/src/main/asciidoc/redis-reference.adoc (hunk 2 above)
Comment (machi1990) [reply docs/src/main/asciidoc/redis-reference.adoc:284]: Yes, the comment was more about adding the Quarkus related code snippets of course with the assumption that ones understand the underlying redis workings

File: extensions/redis-client/runtime/src/main/java/io/quarkus/redis/runtime/datasource/ReactiveTransactionalCountMinCommandsImpl.java (hunk 1 above)
Comment (machi1990) [reply extensions/redis-client/runtime/src/main/java/io/quarkus/redis/runtime/datasource/ReactiveTransactionalCountMinCommandsImpl.java]: Thanks

File: docs/src/main/asciidoc/redis-reference.adoc (hunk 2 above)
Comment (cescoffier) [reply docs/src/main/asciidoc/redis-reference.adoc:284]: I will write something once I'm done with all the redis labs groups (... never ending task :-D)

Comment (machi1990) [reply docs/src/main/asciidoc/redis-reference.adoc:284]: Sounds good. Thanks a lot @cescoffier for the improvements!

**Pull request merged**